Test1_2/SparseVector/SparceVector.cs

```diff
@@ -0,0 +1,270 @@
+                secondElement = secondElement.Next;
+                continue;
+            }
+            if (secondElement.CellName > firstElement.CellName)
+            {
+                newVector[firstElement.CellName] = firstElement.Value;
+                firstElement = firstElement.Next;
+            }
+        }
+        return newVector;
+    }
+
+    public static int ScalarProduct(SparceVector first, SparceVector second)
+    {
+        var scalarProduct = 0;
+        var firstElement = first.head;
+        var secondElement = second.head;
+        while (firstElement != null || secondElement != null)
+        {
+            if (firstElement.CellName == secondElement.CellName)
+            {
+                scalarProduct += firstElement.Value * secondElement.Value;
+                firstElement = firstElement.Next;
+                secondElement = secondElement.Next;
+                continue;
+            }
+            if (firstElement.CellName > secondElement.CellName)
+            {
+                secondElement = secondElement.Next;
+                continue;
+            }
+            if (secondElement.CellName > firstElement.CellName)
+            {
+                firstElement = firstElement.Next;
+            }
+        }
+        return scalarProduct;
+    }
+
+    private void DeleteElement(VectorElement vectorElement)
+    {
+        if (vectorElement == head)
+        {
+            if (vectorElement.Next != null)
+            {
+                head = vectorElement.Next;
+                vectorElement.Next.Previous = null;
+                vectorElement.Next = null;
+            }
+            else
+            {
+                head = null;
+            }
+        }
+        if (vectorElement.Previous != null)
+        {
+            vectorElement.Previous.Next = vectorElement.Next;
+        }
+        if (vectorElement.Next != null)
+        {
+            vectorElement.Next.Previous = vectorElement.Previous;
+        }
+        vectorElement.Next = null;
+        vectorElement.Previous = null;
+    }
+
+    public void PrintVector()
+    {
+        if (IsNull())
+        {
+            Console.WriteLine("Вектор из нулей");
+        }
+        VectorElement currentElement = head;
+        while (currentElement != null)
+        {
+            Console.WriteLine($"{currentElement.CellName} {currentElement.Value}");
+            currentElement = currentElement.Next;
+        }
+    }
+
+    public int GetValue(int name)
+    {
+        var currentElement = head;
+        while (currentElement != null)
+        {
+            if (currentElement.CellName == name)
+            {
+                return currentElement.Value;
+            }
+            currentElement = currentElement.Next;
+        }
+        return 0;
+    }
+
+    public int Length()
+    {
+        return logicalLength;
+    }
+
+    public bool IsNull()
+    {
+        return head == null;
+    }
```

Comment:
Можно это написать лаконичнее:
```suggestion
    public bool IsNull() => head == null;
```